sigv4.go

```diff
@@ -103,29 +125,15 @@ func (rt *sigV4RoundTripper) newBuf() interface{} {
 }
 
 func (rt *sigV4RoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
-	// rt.signer.Sign needs a seekable body, so we replace the body with a
-	// buffered reader filled with the contents of original body.
-	buf := rt.pool.Get().(*bytes.Buffer)
-	defer func() {
-		buf.Reset()
-		rt.pool.Put(buf)
-	}()
-
+	var bodyBytes []byte
```

Comment:
why did you stop using the pool of buffers. this will cause you to allocate slices on every request while before we were reusing buffers across requests.

Reply:
You're right, I lost that in the refactor (As it was mixed with a buffered reader which I think we don't need anymore), reimplemented it :)